example/lib/develop/peripheral_list/peripheral_list_screen.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:io';
+
+import 'package:blemulator_example/develop/model/ble_peripheral.dart';
+import 'package:blemulator_example/develop/peripheral_list/bloc.dart';
+import 'package:blemulator_example/develop/peripheral_list/components/connection_state_view.dart';
+import 'package:blemulator_example/develop/peripheral_list/components/rssi_view.dart';
+import 'package:blemulator_example/develop/util/custom_colors.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PeripheralListScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return _buildTestLayout(context);
+  }
+
+  Widget _buildTestLayout(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Bluetooth peripherals'),
+        // TODO: - replace with a toggle button based on state.scanningEnabled
+        actions: <Widget>[
+          MaterialButton(
+            child: Text('Start scanning'),
+            onPressed: () => _startScanning(context),
+          ),
+          MaterialButton(
+            child: Text('Stop scanning'),
+            onPressed: () => _stopScanning(context),
+          ),
+        ],
+      ),
+      body: Container(
+        child: BlocBuilder<PeripheralListBloc, PeripheralListState>(
+            builder: (context, state) {
+          return ListView.builder(
+            itemCount: state.peripherals.length,
+            itemBuilder: (context, index) {
+              return _buildRow(context, state.peripherals[index]);
+            },
+            padding: EdgeInsets.fromLTRB(16.0, 8.0, 16.0, 8.0),
+          );
+        }),
+      ),
+      backgroundColor: CustomColors.systemGroupedBackground,
+    );
+  }
+
+  static Widget _buildRow(BuildContext context, BlePeripheral peripheral) {
```

Comment:
why is it static? maybe it is worth to extract row widget as a separate widget?

Reply:
Agree with that. The row is going to be extracted as a separate widget.

Reply:
Fixed.